tests/modules/programs/thunderbird/thunderbird.nix

```diff
@@ -62,16 +66,31 @@
       };
     };
 
+    nativeMessagingHosts = with realPkgs;
+      [
+        # NOTE: this is not a real Thunderbird native host module but Firefox; no
+        # native hosts are currently packaged for nixpkgs or elsewhere, so we
+        # have to improvise. Packaging wise, Firefox and Thunderbird native hosts
+        # are identical though. Good news is that the test will still pass as
+        # long as we don't attempt to run the mail client itself with the host.
+        # (Which we don't.)
```

Comment:
One thing this just made me curious about because of this note. If we have a shared file for configuring the native messaging hosts and someone wants to install a firefox native messaging host.... are they going to get errors in thunderbird?

Reply:
You mean whether Thunderbird will attempt to call to native hosts that belong to Firefox? Only when Thunderbird has an extension installed that would refer to this native host. For example here's how `cb_thunderlink` that I use spawns the corresponding native host: https://github.com/CamielBouchier/cb_thunderlink/blob/089490e44919dca6127f0398db65f05474e12050/mail_extension/cb_background.js#L452

I don't think TB and Firefox in general share native hosts; but if they do, each of them will start their own native process to communicate with.

Reply:
If you suggest that the comment shouldn't imply there would be anything wrong if we WOULD run TB in the test suite per se, then yeah maybe we should clean up. I wanted to communicate that home-manager won't care what's in the actual directory. Perhaps I should make it more clear.

Reply:
Alright, thanks. Sorry, not familiar with native messaging hosts really. Sounds like native messaging hosts just present an interface for communication that a plugin/extension has to try to communicate through itself. So we shouldn't just cause issues by their existence in a file system.

Reply:
yeah. it's a way for TB/FF to talk to the system. But since it's generally unsafe, plugins can only do that through native hosts that are prepared (in our case by HM) in a well known directory. If a json file that describes the host module is put in the directory, then TB/FF will allow the plugin to execute it.

I will update the comment, I think it's indeed a bit misleading. (Esp. since as you pointed out we should handle the case of null package.)